sandberg_christy_ndvi.py

```diff
@@ -0,0 +1,700 @@
+    pixel_qa : xarray.core.dataarray
+        landsat pixel_qa layer
+    ndvi_bands : list
+        list of two cropped bands to be used for NDVI calculation
+        
+    """     
+    
+    # get the site name ('site') from site directory name 
+    path_to_site_dir = os.path.dirname(os.path.dirname(
+        os.path.normpath(path_to_scene_dir)))
+    site = os.path.basename(os.path.normpath(path_to_site_dir))
+    
+    # get the data acquisition date ('date') from scene directory name 
+    date = os.path.basename(os.path.normpath(path_to_scene_dir))[10:18]
+      
+    # get shapefile needed for 'crop_shape'
+    crop_shape_path = glob(os.path.join(path_to_site_dir, 'vector', '*.shp'))
+    crop_shape = gpd.read_file(os.path.normpath(crop_shape_path[0])) 
+                    
+    # create cloud mask layer from qa band
+    # open the qa band needed for the cloud mask, clip to crop_shape
+    pixel_qa_band_path = glob(os.path.join(path_to_scene_dir, "*qa.tif"))[0]
+    pixel_qa = rxr.open_rasterio(
+        pixel_qa_band_path, masked=True).rio.clip(
+        crop_shape.geometry, from_disk=True).squeeze()
+    
+    # get 'ndvi_bands' list; info needed to calculate NDVI at later step
+    # open the red and infrared bands and clip to crop_shape
+    ndvi_bands_path = sorted(
+        glob(os.path.join(path_to_scene_dir, "*band[4-5]*")))
+    ndvi_bands = []
+    for band_path in ndvi_bands_path:
+        band = rxr.open_rasterio(band_path, masked=True).rio.clip(
+            crop_shape.geometry, from_disk=True).squeeze()
+        
+        # mask values to valid range (if valid_range is given)
+        if valid_range:
+            mask = ((band < valid_range[0]) | (band > valid_range[1]))
+            clean_band = band.where(~xr.where(mask, True, False)) 
+        ndvi_bands.append(clean_band)
+    
+    # function returns four variables
+    return(site, date, pixel_qa, ndvi_bands)
+    
+    
+#  function to calculate NDVI, with data removed when it is hidden by clouds 
+def calculate_masked_ndvi(ndvi_bands, pixel_qa):
+    """calculates NDVI; removes data that is hidden by clouds 
+    
+    Calculates normalized difference from two arrays of same shape. Math will 
+    be calculated (b1-b2) / (b1+b2). Also removes values that are obscured by 
+    clouds, applying the pixel_qa layer when it's values are found in the 
+    'cloud_vals' list maintained inside this function.
+    
+     Parameters
+     ----------
+     ndvi_bands : list 
+         List of two numpy arrays of same shape
+     pixel_qa : xarray.core.dataarray
+         landsat pixel_qa layer
+    
+    Returns
+     ----------
+     ndvi_masked : numpy array
+         The element-wise result of (b1-b2) / (b1+b2) calculation after the 
+         cloud mask is applied 
+
+    """
+    ndvi = (ndvi_bands[1]-ndvi_bands[0]) / (ndvi_bands[1]+ndvi_bands[0])
+    
+    cloud_vals = [328, 392, 840, 904, 1350, 352, 368, 416, 432, 480, 864, 
```

Comment:
@csandberg303   If you want to use the `import earthpy.mask as em` package, instead of just enumerating the 'cloud_vals', you could use that package in the following way:

```
# Create list objects of pixel qa raster values to mask cloud effects and combine into single list
high_cloud_confidence = em.pixel_flags["pixel_qa"]["L8"]["High Cloud Confidence"]
cloud = em.pixel_flags["pixel_qa"]["L8"]["Cloud"]
cloud_shadow = em.pixel_flags["pixel_qa"]["L8"]["Cloud Shadow"]

cloud_vals = cloud_shadow + cloud + high_cloud_confidence
```